pkg/virt-controller/watch/vm_initializer.go

```diff
@@ -0,0 +1,167 @@
+package watch
+
+import (
+	"fmt"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	kubev1 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+)
+
+type VMInitializer struct {
+	clientset  kubecli.KubevirtClient
+	restClient *rest.RESTClient
+	queue      workqueue.RateLimitingInterface
+	store      cache.Store
+}
+
+// FIXME: Both the restClient and clientSet are probably not needed.
+func NewVMInitializer(restClient *rest.RESTClient, queue workqueue.RateLimitingInterface, vmCache cache.Store, clientset kubecli.KubevirtClient) *VMInitializer {
```

Comment:
Since the VM initializer controller and the VM controller are both working on VM objects, what would you think about merging the initializer logic into the VM controller?  That would let them share the same informer and cache.